packages/compiler/src/shadow_css.ts

```diff
@@ -713,7 +713,7 @@ export class ShadowCss {
       // (ie: ".\fc ber" for ".über") is not a separator between 2 selectors
       // also keep in mind that backslashes are replaced by a placeholder by SafeSelector
       // These escaped selectors happen for example when esbuild runs with optimization.minify.
-      if (part.match(_placeholderRe) && selector[res.index + 1]?.match(/[a-fA-F\d]/)) {
+      if (part.match(/__esc-ph-(\d+)__/g) && selector[res.index + 1]?.match(/[a-fA-F\d]/)) {
```

Comment:
I don't think the capturing group is needed here, nor is the global flag:

```suggestion
      if (part.match(/__esc-ph-\d+__/) && selector[res.index + 1]?.match(/[a-fA-F\d]/)) {
```

Without the global flag this regex can be safely hoisted to stored in a top-level variable.

Reply:
Does this work when `part` contains both an escaped pseudo class _and_ an escaped backslash? The regex test is across all of `part`, so wouldn't the escaped backslash escaping then still cause this logic to be triggered for the pseudo occurrence, resulting in the same issue as before? I may be missing something here, but it'd at least be interesting to have a test for such case.

Reply:
Yep I can confirm that a selector like `:host.pr\\fc fung` isn't handled correctly.

Reply:
But `.one\\:\\fc ber {}` is handled correctly I'm adding it in the tests.
 
I won't fix the :host with an escaped selector, it will be hard to fix.